File: langs/php/src/Dotlyte.php
<?php

declare(strict_types=1);

namespace Dotlyte;

/**
 * Main entry point for DOTLYTE configuration loading.
 */
final class Dotlyte
{
    /**
     * Load configuration from all available sources.
     *
     * @param LoadOptions|null $options
     * @return Config
     */
    public static function load(?LoadOptions $options = null): Config
    {
        $options ??= new LoadOptions();
        $loader = new Loader($options);
        return $loader->load();
    }
}

File: langs/java/src/main/java/dev/dotlyte/Dotlyte.java
package dev.dotlyte;

/**
 * Entry point for the DOTLYTE configuration library.
 *
 * <pre>{@code
 * Config config = Dotlyte.load();
 * int port = config.getInt("port", 3000);
 * String host = config.get("database.host", "localhost");
 * }</pre>
 */
public final class Dotlyte {

    private Dotlyte() {
        // Utility class
    }

    /**
     * Load configuration with default options.
     *
     * @return a Config object with merged configuration
     */
    public static Config load() {
        return load(LoadOptions.builder().build());
    }

    /**
     * Load configuration with custom options.
     *
     * @param options the loading options
     * @return a Config object with merged configuration
     */
    public static Config load(LoadOptions options) {
        return new Loader(options).load();
    }
}
